pulsar-client/src/main/java/org/apache/pulsar/client/impl/AbstractBatchMessageContainer.java

```diff
@@ -46,6 +47,8 @@ public abstract class AbstractBatchMessageContainer implements BatchMessageConta
     protected long currentTxnidLeastBits = -1L;
 
     protected static final int INITIAL_BATCH_BUFFER_SIZE = 1024;
+    protected static final int SHRINK_COOLING_OFF_PERIOD = 10;
+    protected int consecutiveShrinkTime = 0;
```

Comment:
Shall these two fields private fields in `BatchMessageContainerImpl`?

Reply:
OK, private fields looks better for `allocator` is private